source/common/router/config_impl.h

```diff
@@ -860,6 +872,18 @@ class RouteEntryImplBase : public RouteEntry,
     envoy::type::v3::FractionalPercent fractional_runtime_default_{};
   };
 
+  struct RouteStatsContext {
+    RouteStatsConfigSharedPtr route_stats_config_;
```

Comment:
I have moved rest of the usages to `RouteStatsConfigOptConstRef` and changed this to unique ptr.  Having RouteStatsConfig as is (Without inlining in RouteContext) is useful because for rest of the code that uses RouteStatsConfig, we can just return it as is.

Reply:
Sorry I meant inline into RouteStatsContext, not RouteStats. I'm unclear on the value of having RouteStatsContext and RouteStatsConfigureSharedPtr as distinct structures -- they are both new to this PR so I don't think it would make the PR bigger to tie them together.

Reply:
> Sorry I meant inline into RouteStatsContext, not RouteStats

Yes. That is what I mean too

> . I'm unclear on the value of having RouteStatsContext and RouteStatsConfigureSharedPtr as distinct structure

We can certainly combine. My point was rest of the code that generates stats, does not need all the data except what is there in RouteStatsConfig - So the users do not need any other data. If you think it better to combine, I can do it - not worried about PR getting bigger

Reply:
I have eliminated one, PTAL